dhis-2/dhis-services/dhis-service-analytics/src/test/java/org/hisp/dhis/analytics/data/AnalyticsServiceSubExpressionTest.java

```diff
@@ -0,0 +1,299 @@
+
+        ouA = createOrganisationUnit( 'A' );
+        ouB = createOrganisationUnit( 'B', ouA );
+        ouC = createOrganisationUnit( 'C', ouA );
+        organisationUnitService.addOrganisationUnit( ouA );
+        organisationUnitService.addOrganisationUnit( ouB );
+        organisationUnitService.addOrganisationUnit( ouC );
+
+        CategoryOption optionA = new CategoryOption( "CategoryOptionA" );
+        CategoryOption optionB = new CategoryOption( "CategoryOptionB" );
+        categoryService.addCategoryOption( optionA );
+        categoryService.addCategoryOption( optionB );
+
+        Category categoryA = createCategory( 'A', optionA, optionB );
+        categoryService.addCategory( categoryA );
+
+        CategoryCombo categoryComboA = createCategoryCombo( 'A', categoryA );
+        categoryService.addCategoryCombo( categoryComboA );
+
+        CategoryOptionCombo cocA = createCategoryOptionCombo( categoryComboA, optionA );
+        CategoryOptionCombo cocB = createCategoryOptionCombo( categoryComboA, optionB );
+        cocA.setUid( "OptionCombA" );
+        cocB.setUid( "OptionCombB" );
+        categoryService.addCategoryOptionCombo( cocA );
+        categoryService.addCategoryOptionCombo( cocB );
+        CategoryOptionCombo aocA = categoryService.getDefaultCategoryOptionCombo();
+
+        categoryComboA.getOptionCombos().add( cocA );
+        categoryComboA.getOptionCombos().add( cocB );
+        categoryService.updateCategoryCombo( categoryComboA );
+
+        IndicatorType indicatorTypeA = createIndicatorType( 'A' );
+        indicatorTypeA.setFactor( 1 );
+        indicatorService.addIndicatorType( indicatorTypeA );
+
+        indicatorA = createIndicator( 'A', indicatorTypeA );
+        indicatorA.setNumerator( "1" ); // to be overwritten
+        indicatorA.setDenominator( "1" );
+        indicatorService.addIndicator( indicatorA );
+
+        dataValueService.addDataValue( newDataValue( deA, perA, ouB, cocA, aocA, "1" ) );
+        dataValueService.addDataValue( newDataValue( deA, perA, ouC, cocB, aocA, "2" ) );
+        dataValueService.addDataValue( newDataValue( deB, perA, ouB, cocA, aocA, "B" ) );
+        dataValueService.addDataValue( newDataValue( deB, perA, ouC, cocB, aocA, "C" ) );
+
+        // Wait before generating analytics tables
+        Thread.sleep( 1000 );
+
+        // Generate analytics tables
+        analyticsTableGenerator.generateTables( AnalyticsTableUpdateParams.newBuilder().build(),
+            NoopJobProgress.INSTANCE );
+
+        // Wait after generating analytics tables
+        Thread.sleep( 1000 );
+    }
+
+    @Override
+    public void tearDownTest()
+    {
+        for ( AnalyticsTableService service : analyticsTableServices )
+        {
+            service.dropTables();
+        }
+    }
+
+    // -------------------------------------------------------------------------
+    // Test
+    // -------------------------------------------------------------------------
+
+    @Disabled
```

Comment:
?

Reply:
See my comments under Testing (and also the several commits "try to fix integration test"). This works fine on my system in the IDE or on the command line, but doesn't work on the integration test server. I didn't want to further hold up the PR after my attempts at fixing this had failed. (Insights from anyone would be most welcome!) I've now put a comment in the code about this.

Reply:
Missed that when speed reading the description :D